WalletWasabi/Services/LegalChecker.cs

```diff
@@ -0,0 +1,119 @@
+using Nito.AsyncEx;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Legal;
+using WalletWasabi.Logging;
+using WalletWasabi.Models;
+
+namespace WalletWasabi.Services
+{
+	public class LegalChecker : IDisposable
+	{
+		public const string LegalFolderName = "Legal";
+		public const string ProvisionalLegalFolderName = "Provisional";
+
+		private bool _disposedValue;
+
+		public LegalChecker(string dataDir)
+		{
+			LegalFolder = Path.Combine(dataDir, LegalFolderName);
+			ProvisionalLegalFolder = Path.Combine(LegalFolder, ProvisionalLegalFolderName);
+		}
+
+		private AsyncLock LegalDocumentLock { get; } = new();
```

Comment:
I would add comment saying what needs to be locked by the lock.

`/// <remarks>Lock object to guard <see cref="SOMETHING"/> property.</remarks>`

Reply:
ACK.